### PR TITLE
Pattern Library: Convert to Bower Package

### PR DESCRIPTION
**Pull request by talbs**
This work converts the``_src/pattern-library`` into a bower package, which includes the following work:

* update ``bower.json`` in prep for packaging
* update directory structure - remove unnecessary + PL Doc site files
* add official ``LICENSE``, and ``AUTHOR`` files to repo
* add a ``CHANGELOG``
* updating ``README`` for accuracy, new workflows, and links to doc.

- - -

Note, this PR is 1 of 3 in a series that needs to be merged around the same time:

1. **This PR** - Configure PL as a Bower Package
2. Registering ``edx-pattern-library`` as a public Bower Package (see http://bower.io/docs/creating-packages/#register)
3. **https://github.com/edx/ux-pattern-library/pull/146** - Updating the ``pldoc`` (Documentation Site's new master working branch) to use registered Bower Package rather than commit hash
4. **https://github.com/edx/ux-pattern-library/pull/142** - Publish ``pldoc`` branch to ``gh-pages``

- - -

#### Reviewers

* [x] **FED/Package Infrastructure** - @dsjen 
* [ ] **README/Usage Documentation** - @marcotuts and @clrux 
* [x] **Open edX License and Contributing Standards** - @sarina

### CODE REVIEW

**Comment by talbs** [comment]
@marcotuts and @clrux, here are some quick links to the new/updated docs:

* README - https://github.com/edx/ux-pattern-library/blob/talbs/use-bower/README.md
* CONTRIBUTING - https://github.com/edx/ux-pattern-library/blob/talbs/use-bower/CONTRIBUTING.md

**Comment by talbs** [comment]
@jbarciauskas, apologies for the random ping, here are the files and details I could use your eyes on:

* **[CONTRIBUTING](https://github.com/edx/ux-pattern-library/blob/talbs/use-bower/CONTRIBUTING.md)** - I used the edx-platform repo's copy of this file as a basis, but a fair amount of the communication infrastructure and contribution workflow wasn't applicable to this repo/work yet. We don't have an IRC channel, nor do we have any JIRA hooks, so 1) Github Issues + emailing the UX Team (we're directly managing this codebase) for communication and 2) UX Design and Front End reviews for workflow seemed more reasonable and fitting.
* **[LICENSE](https://github.com/edx/ux-pattern-library/blob/talbs/use-bower/LICENSE.TXT)** - copied straight from the xblock repo per our conversation. Just make sure I didn't goof anything here, please.
* **[README](https://github.com/edx/ux-pattern-library/blob/talbs/use-bower/README.md)** - If you have time, a pair of eyes on the doormat doc would be great to make sure it generally makes sense.

**Comment by clrux** [comment]
@talbs I was actually looking at this last night and had a few comments on the README:

* I'm wondering if going into detail about `gulp` aliasing is necessary as the instructions pretty much override the `gulp` instance on the entire system. If a developer has more than one `gulp` project in play, this would make things fonky. I recommend removing the bit about aliasing, letting the developers do what they feel is best.
* When running `bower install` I saw two dependencies with "invalid meta": https://gist.github.com/clrux/730bd08a0a53f5811f26

Contributions looks good. Nice work here.

**Comment by jbarciauskas** [comment]
@talbs @sarina do we want to use GitHub Issues? Isn't that asking for trouble, in terms of tracking work in two different places?

**Comment by talbs** [comment]
> @talbs @sarina do we want to use GitHub Issues? Isn't that asking for trouble, in terms of tracking work in two different places?

@jbarciauskas, a good point! I'm happy to try to go with current workflows, places of communication, and work tracking as this is an edX/Open edX project. I am however a bit concerned about how much overhead some of the platform-level mechanism would cost our team for this smaller project.

Some initials ways to avoid GH issues and sync up workflows might include:

1. Make a particular issue type under Open Source Pull Requests for the Pattern Library in the current Open Source JIRA ticketing workflow
2. Or make our UX Pattern library epic (https://openedx.atlassian.net/browse/UX-1396) public and let folks see the current bugs and planned stories, as well as create their own issues for triage.

What do you and @sarina think?

**Comment by talbs** [comment]
> @talbs I was actually looking at this last night and had a few comments on the README:

> I'm wondering if going into detail about gulp aliasing is necessary as the instructions pretty much override the gulp instance on the entire system. If a developer has more than one gulp project in play, this would make things fonky. I recommend removing the bit about aliasing, letting the developers do what they feel is best.

@clrux, very fair point, we can leave that as an in-team pro tip for us. I'll remove that bit and simplify that area of instruction. Thanks!

- - -

> 
When running bower install I saw two dependencies with "invalid meta": https://gist.github.com/clrux/730bd08a0a53f5811f26

@clrux, yeah, looks like bi-app's bower package is missing a few things. Regardless, that's not our code and the package is loaded successfully even if Bower grumbles a bit. I'll look at bi-app's repo to see if the issue is known. If not, i'll file an issue and give you props. Good eyes!

**Comment by talbs** [comment]
@clrux, removed that gulp alias line in https://github.com/edx/ux-pattern-library/commit/0b85f52f1aee34962d2a71f6f6d764b0f7a9fe6a.

**Comment by clrux** [comment]
Cool, looks good. :+1:

**Comment by sarina** [comment]
> do we want to use GitHub Issues?

No

Also I would love to avoid a custom new workflow for this library. I don't
see a point. Let's sync up in person.

File: README.md [hunk 1]
```diff
@@ -64,86 +83,135 @@ npm install
...
-* `watch` will start watching source files and will re-run the appropriate tasks when those files change.
 
 After running the default task, your local instance of the Pattern Library can be viewable at http://localhost:3000. There are additional views:
 
 * http://localhost:3000/examples/**/* - where full page examples and demos are kept
 * http://localhost:3000/elements/**/* - permalinks for individually documented pattern library elements
 
-**NOTE: some gulp tasks had to be separated to support both the pattern library 1)elements and 2)reference site UI separately.** Any task prefixed with `pldoc_*` refers to the reference site UI and not the elements.
+## Development - UX Pattern Library Package
+Since the ``master`` branch only contains the bower package source code, its a bit tricky to develop locally with an accurate sense of what one's code changes will affect. With that in mind, its recommended that you work with the ``pldoc`` branch or another edX/Open edX app injesting the pattern library source by:
```
Comment (dsjen) [inline README.md]: spelling: it's (instead of its)

File: README.md [hunk 2]
```diff
@@ -64,86 +83,135 @@ npm install
 
 This runs through all dependencies listed in package.json and downloads them to a node_modules folder in your project directory.
 
-**NOTE: You may need to run this more than just once (if encountering errors when running the gulp command) as new dependencies may have been added**
+**NOTE:** You may need to run this more than just once (if encountering errors when running the gulp command) as new dependencies may have been added
```
Comment (dsjen) [inline README.md]: Looks to be missing a period at the end.

File: README.md [hunk 3]
```diff
@@ -64,86 +83,135 @@ npm install
...
-* Brian Talbot (@talbs)
-* Frances Botsford (@frrrances)
-* Marco Morales (@marcotuts)
-* Tyler Nickerson (@nickersoft)
+### Step 4: Configure Settings
+There are a few places you can start to explore configuring aspects of the UX Pattern Library for your purposes. You can find many configurations and base settings in ``src/sass/utilities/_variables.scss`` and can override/customize those in your application's ``utilities.scss`` partial (they'll be used throughout the rest of the compile).
+
+**NOTE:** The code and design contained in the library has not been tested for use alongside other UI Frameworks such as Bootstrap or Foundation.
+
+**NOTE:** The source code of the Pattern Library does not include any local or production-focused Sass/Front End tooling. Its expected that your app/site has its own Sass compiler (preferrably LibSass).
```
Comment (dsjen) [inline README.md]: The first "Its" should be "it's" instead.

File: bower.json [hunk 4]
```diff
@@ -1,12 +1,23 @@
...
+  "keywords": [
+    "edx",
+    "patterns",
+    "atomic design",
+    "pattern library",
+    "UI starter"
+  ],
+  "main": "_variables.scss",
+  "license": "Apache 2.0",
   "private": true,
```
Comment (dsjen) [inline bower.json]: Is this this private?

File: README.md [hunk 5]
```diff
@@ -1,24 +1,37 @@
...
+To work on the code of the documentation site portion of this repo, you'll need the following things installed on your local machine.
 
 1. [Jekyll](http://jekyllrb.com/)
 2. [NodeJS](http://nodejs.org/)
 3. [Bower](http://bower.io/)
 4. [GulpJS](https://github.com/gulpjs/gulp) + other gulp-based packages
 
-### Install Jekyll
+### Step 0: Switch to the ``pldoc`` Branch
+The master version of this repo is dedicated to the bower package version of the UX Pattern Library, but the documentation site is hosted through Github Pages and the [``gh-pages`` branch](http://whatsock.com/training/matrices/visual-aria.htm). Development of the documentation site is however handled by the ``pldoc`` branch. When its time to publish, PRs are made from this main development branch against ``gh-pages``
```
Comment (dsjen) [inline README.md]: "its" should be "it's" instead.

File: README.md [hunk 6]
```diff
@@ -53,9 +59,22 @@ After Bower is installed, you can install all of the edX UX Pattern Library depe
 bower install
 ```
 
-NOTE: Bower packages are installed to the ``_src/components`` directory. These files should not be edited or removed outside of the Bower workflow.
+**NOTE**: Bower files should not be edited or removed outside of the Bower workflow.
+
+- - -
+
+### Step 3: Install NodeJS
```
Comment (dsjen) [inline README.md]: Does node need to be installed before bower?

File: bower.json [hunk 7]
```diff
@@ -1,12 +1,23 @@
 {
-  "name": "edx-ux-pattern-library",
-  "version": "0.0.1",
+  "name": "edx-pattern-library",
+  "version": "0.1.0",
```
Comment (dsjen) [inline bower.json]: Is this version 0.2 now?

**Comment by dsjen** [comment]
I have a few small questions.  The most pressing is the order of installs--whether nodejs needs to be installed before bower.  I think it does, but I could be wrong.  Otherwise, LGTM! :+1:

File: README.md (hunk 6 above)
Comment (talbs) [inline README.md]: Amazeballs catch! Yes, it does. I've switched the order of instructions in https://github.com/edx/ux-pattern-library/commit/72c2a49a037c4d863862af0dc159b0cfc6e86bdd. Thanks!

**Comment by talbs** [comment]
Thanks for the really thorough review @dsjen. I addressed all of the 5th grade grammar issues you found and synced up the ``bower.json`` details to make the package public and version 0.2.0.

**Comment by talbs** [comment]
@sarina, thanks for the thoughts here as well as when chatting in-person. As we discussed, I'm all for:

* syncing up Pattern Library contributions workflow as much with the general platform/Open edX contribution process.
* relying on the existing CONTRIBUTING.rst file in the edx-platform repo (alongside any caveats or extra details for the UX Pattern Library).

I've tweaked some of the doc in this PR with those bits in mind (https://github.com/edx/ux-pattern-library/commit/758ebc838cb9a04d4a5fe7ceccf545e4ec434e45). I could use some help setting up any bots, JIRA projects, bug boards, etc. to link from the updated README file in this branch.

Let me know when you have time and if you guys have any other concerns.

**Comment by talbs** [comment]
@clrux, no issue existed, so I made one on bi-app-sass's GH repo - https://github.com/anasnakawa/bi-app-sass/issues/12.

**Comment by talbs** [comment]
Update: Thanks to @sarina, we now have a few things in place: 

* an updated README that contains general Open edX contribution references/workflow as well as details on Pattern Library specifics and JIRA project workflow - https://github.com/edx/ux-pattern-library/blob/talbs/use-bower/README.md#contributions
* a new edX UX Pattern library JIRA Project that is open and can receive issues from the community - https://openedx.atlassian.net/projects/UXPL/issues

File: README.md [hunk 8]
```diff
@@ -1,24 +1,51 @@
 # edX Pattern Library
-The (working) Visual, UI, and Front End Styleguide for edX.
+The (working) UI library and Front End Styleguide for edX/Open edX applications and sites.
+
+- - -
+
+## Table of Contents
+
+1. [Overview](#overview)
+2. [Licence](#license)
```
Comment (sarina) [inline README.md]: License (with an S not a C)

File: README.md [hunk 9]
```diff
@@ -1,24 +1,51 @@
...
 This library contains the following:
 
-* [A working preview and documentation system for edX application UI](http://ux.edx.org)
+* [A working preview and documentation system for edX application UI](http://ux.edx.org) - managed through the [``pldoc``](https://github.com/edx/ux-pattern-library/tree/pldoc) branch
 * Styleguides and standards for [general Front End](https://github.com/edx/ux-pattern-library/wiki/Styleguide:-General), [HTML](https://github.com/edx/ux-pattern-library/wiki/Styleguide:-HTML), [Sass/CSS](https://github.com/edx/ux-pattern-library/wiki/Styleguide:-Sass-&-CSS), and [Accessibility-minded](https://github.com/edx/ux-pattern-library/wiki/Styleguide:-Accessibility) development
-* TODO: Portable Sass/CSS modules for use within edX applications
+* Portable Sass/CSS utilities and modules for use within edX applications  - managed through the [``master``](https://github.com/edx/ux-pattern-library/tree/master) branch
 
-- - -
+## Licence
```
Comment (sarina) [inline README.md]: License (with an S not a C)

File: README.md [hunk 10]
```diff
@@ -1,24 +1,51 @@
...
+The code in this repository is licensed the Apache 2.0 license unless otherwise noted. Please see the [LICENSE file](https://github.com/edx/ux-pattern-library/blob/master/LICENCE.txt) for details.
```
Comment (sarina) [inline README.md]: should be LICENSE.txt (S not C)

File: README.md (hunk 8 above)
Comment (sarina) [inline README.md]: Overall I would recommend running a spell check over the README

File: README.md [hunk 11]
```diff
@@ -64,86 +97,165 @@ npm install
...
+## Contributions
+Contributions are very welcome. The easiest way is to fork this repo, and then make a pull request from your fork. The first time you make a pull request, you may be asked to sign a Contributor Agreement.
+
+Please refer to our [contributor guidelines](https://github.com/edx/edx-platform/blob/master/CONTRIBUTING.rst) in the main edx-platform repo for important additional information.
+
+### Contributing and the UX Pattern Library
+There are a few additional details alongside our general guidelines to keep in mind contributing to the UX Pattern Library:
+
+#### Pattern Library Features, Ideas, and Improvements
+If you're looking to suggest an idea or you're thinking about developing a feature, start a discussion [by visiting the Open edX JIRA site](https://openedx.atlassian.net/secure/Dashboard.jspa) and  create a new "Issue" by selecting the "Create" button at the top of the page. Choose the project "UX Pattern Library" and the issue type "New Feature" or "Improvement" (you may first need to [create a free JIRA account](https://openedx.atlassian.net/admin/users/sign-up)).
```
Comment (sarina) [inline README.md]: You should consider linewrapping this file so future changes are easy to see in diff form.

**Comment by sarina** [comment]
:+1: from me

File: README.md (hunk 11 above)
Comment (talbs) [inline README.md]: Will do, thx much!

**Comment by talbs** [comment]
Hi, all.

This direction (and its subsequent PRs) would cause a huge shift in workflow for Pattern Library development. This would require us (and perhaps more importantly, any contributors) to work in strange and slow ways locally all for the benefit of packaging.

**After talking with a few of you as well as @frrrances and @singingwolfboy, I'm closing this line of PRs in favor of https://github.com/edx/ux-pattern-library/pull/152**, which preps the ``pattern-library`` source code as a Bower package while configuring bower to ignore the doc site (PLDOC) and other extraneous code in the repo, which allows us to design/develop new UI patterns within the isolated and locally friendly PLDOC site/workflow. 

@sarina, all of the doc changes and org you've helped with have been ported over, so I'm leaving your review as complete. Shout if you have concerns or want to revisit

@dsjen, the same goes for you - none of the ``bower.json`` metadata has been altered (aside from adding PLDOC files to the package's ignore list).

**Pull request closed without merging**